legacy/reqcounter/reqcounter.go

```diff
@@ -19,76 +19,127 @@ package reqcounter
+func (rc *RequestCounter) Flush() {
 	// Hold onto the lock when reading & writing the counter.
-	rc.mutex.Lock()
-	defer rc.mutex.Unlock()
+	rc.Mutex.Lock()
+	defer rc.Mutex.Unlock()
 
 	// Log the number of requests within this measurement window.
-	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", counter.since.Format("2006-01-02 15:04:05"), rc.requests)
-	logrus.Debug(msg)
+	msg := fmt.Sprintf("From %s to %s [%d min] there have been %d requests to GCR.", rc.Since.Format(loggingFormat), time.Now().Format(loggingFormat), rc.Interval/time.Minute, rc.Requests)
+	Debug(msg)
 
 	// Reset the counter.
-	rc.requests = 0
-	rc.since = time.Now()
+	rc.reset()
+}
+
+// reset clears the request counter and stamps the current time of reset.
+// TODO: @tylerferrara in the future, use the request counter field 'persistent' to check
+// if the counter should be reset.
+func (rc *RequestCounter) reset() {
+	rc.Requests = 0
+	rc.Since = time.Now()
+}
+
+// watch continuously logs the counter at the specified intervals.
+func (rc *RequestCounter) watch() {
+	go func() {
+		for {
+			time.Sleep(rc.Interval)
+			rc.Flush()
+		}
+	}()
+}
+
+// NetworkMonitor is the primary means of monitoring network traffic between CIP and GCR.
+type NetworkMonitor struct {
+	Counters []*RequestCounter
+}
+
+// increment adds 1 to each request counter, signifying a new request has been made to GCR.
+func (nm *NetworkMonitor) increment() {
+	// Increment each counter.
+	for _, rc := range nm.Counters {
+		rc.Increment()
+	}
+}
+
+// Log begins logging each counter at their specified intervals.
+func (nm *NetworkMonitor) Log() {
+	for _, rc := range nm.Counters {
+		rc.watch()
+	}
 }
 
 const (
-	// measurementWindow specifies the length of time to wait before logging the RequestCounter. Since Google's
+	// MeasurementWindow specifies the length of time to wait before logging the RequestCounter. Since Google's
 	// Container Registry specifies a quota of 50,000 HTTP requests per 10 min, the window
 	// for recording requests is set to 10 min.
 	// Source: https://cloud.google.com/container-registry/quotas
-	measurementWindow = time.Minute * 10
+	MeasurementWindow = time.Minute * 10
+	// loggingFormat specifies the syntax for logging time stamps of request counters.
+	loggingFormat = "2006-01-02 15:04:05"
 )
 
 var (
-	// enableCounting will only become true if the Init function is called. This allows
+	// EnableCounting will only become true if the Init function is called. This allows
 	// requests to be counted and logged.
-	enableCounting = false
-	// counter will continuously be modified by the Increment function to count all
-	// HTTP requests to GCR.
-	counter = &RequestCounter{}
+	EnableCounting bool
+	// Monitor holds all counters for recording HTTP requests to GCR.
+	Monitor *NetworkMonitor
+	// Debug is defined to simplify testing of logrus.Debug calls.
+	// type: func(args ...interface{})
```

Comment:
Please instead instead define the type with the `type` keyword, not as a comment.

Reply:
I forgot that was even possible! Fixing.